npm/src/clientfileprocessing/index.ts

```diff
@@ -23,7 +23,21 @@ export class ClientFileProcessing {
     this.s3Upload = s3Upload
   }
 
-  progressCallback(progressInformation: IProgressInformation) {
+  /**
+   * Updates a progress bar with a given percentage
+   * @param percent The percentage to update the progress bar with  of type {@link Number}
+   */
+  updateProgressBarPercentage(percent: Number) {
+    const progressBarElement: HTMLDivElement | null = document.querySelector(
+      ".progress-display"
+    )
+
+    if (progressBarElement) {
+      progressBarElement.setAttribute("value", percent.toString())
+    }
+  }
+
+  progressBarSwitcher = (progressInformation: IProgressInformation) => {
```

Comment:
I would refactor this function to something like this. You can then set the max percent before calling the two stages that we are showing the percentage processed for:
1. Client file metadata upload
2. File upload

```
/**
   * Displays the progress bar setting the maximum percentage
   * @param maxPercent The maximum percentage to display on progress bar of type {@link Number}
   */
  setProgressBarDisplay(maxPercent: Number) {
    const fileUpload: HTMLDivElement | null = document.querySelector(
      "#file-upload"
    )
    const progressBar: HTMLDivElement | null = document.querySelector(
      "#progress-bar"
    )
    const progressBarElement: HTMLDivElement | null = document.querySelector(
      ".progress-display"
    )

    if (fileUpload && progressBar && progressBarElement) {
      fileUpload.classList.add("hide")
      progressBar.classList.remove("hide")
      progressBarElement.setAttribute("max", maxPercent.toString())
    }
  }
```